src/Microsoft.DotNet.Interactive.Http.Parsing/Parsing/HttpRequestParser.cs

```diff
@@ -448,6 +448,10 @@ private void ConsumeCurrentTokenInto(HttpSyntaxNode node)
                         node = new HttpUrlNode(_sourceText, _syntaxTree);
                         node.Add(ParseEmbeddedExpression());
                     }
+                    else if (CurrentToken is { Kind: HttpTokenKind.Punctuation })
```

Comment:
So far the above two conditions only represented the positive cases that could start a url. The new third condition is for an error case.

Ideally the parser should not return a url node for something that is not a url - it seems odd that the parser would parse a non-url as a url just to improve the error reporting at bind time...

Reply:
Well, the rationale is that the text is there where a url should be, the same way that a token that is interpreted as a word but is completely invalid could be in that location and still create a node. I also prefer this approach so that we have the diagnostic on the url node itself as opposed to the root node